### ISSUE TITLE
ENOTDIR when `init-templates` contains unexpected files

### ISSUE THREAD

**Issue by whilp** [aws/aws-cdk#8382]
If `init-templates` contains unexpected items, the `cdk` CLI will fail with a very mysterious error.

### Reproduction Steps

```
$ touch node_modules/aws-cdk/lib/init-templates/foo
$ cdk help
```

### Error Log

```
(node:60196) UnhandledPromiseRejectionWarning: Error: ENOTDIR: not a directory, scandir 'node_modules/aws-cdk/lib/init-templates/foo'
(node:60196) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:60196) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

### Environment

  - **CLI Version      :** 1.42.0
  - **Framework Version:** 1.42.0
  - **Node.js Version:** 12.13.0
  - **OS               :** Ubuntu
  - **Language (Version):** TypeScript (3.8.3)

### Other

The smallest fix here may be to check that `templateName` is a directory here:

https://github.com/aws/aws-cdk/blob/8aca8558c77aac279ee4cc660a7223d7f8f10c64/packages/aws-cdk/lib/init.ts#L203

This case is not easy to debug because the error occurs very early in the CLI's initialization. `initCommandLine` calls `parseCommandLineArguments` immediately, which in turn immediately awaits the promise of `availableInitLanguages`.

https://github.com/aws/aws-cdk/blob/495c082b0cf9ab65830df4d25b9ecee687d325ca/packages/aws-cdk/bin/cdk.ts#L37

https://github.com/aws/aws-cdk/blob/495c082b0cf9ab65830df4d25b9ecee687d325ca/packages/aws-cdk/bin/cdk.ts#L122 

To resolve that promise, we need to list the contents of the `init-templates` directory and then list the contents of the directories we find there.

https://github.com/aws/aws-cdk/blob/495c082b0cf9ab65830df4d25b9ecee687d325ca/packages/aws-cdk/lib/init.ts#L198-L217

`listDirectory` knows to avoid entries starting with `.`, but will otherwise return everything it sees.

https://github.com/aws/aws-cdk/blob/495c082b0cf9ab65830df4d25b9ecee687d325ca/packages/aws-cdk/lib/init.ts#L222-L226

--- 

This is :bug: Bug Report

**Comment by shivlaks** [comment]
@whilp - can you help me with a minimal repro of the user experience where this would happen.

i.e. what's the circumstance where init-templates will contain an unexpected file? Is the issue with the contents inside of init-templates today or with manipulating it inside of `node_modules`?

**Comment by whilp** [comment]
This issue occurs when other software adds files to CDK’s `init-templates` directory.

We run the CDK CLI using a wrapper script as part of our build/deploy automation. That wrapper (and our entire codebase) happens to be built and managed using Bazel. Until recently, that wrapper was written in bash and invoked as a Bazel `sh_binary`. We discovered this issue when we ported the script to Python and invoked it as a Bazel `py_binary`.

`py_binary` creates `__init__.py` files when it sets up the temporary execution tree for a script. But it does not distinguish between true Python packages, other directories, or (in this case) the `node_modules` tree. Once we discovered the interaction described in this issue, we were able to control this behavior in Bazel to avoid the issue; see the `legacy_create_init` attribute on `py_binary` and the `--incompatible_default_to_explicit_init_py` option in the docs:

https://docs.bazel.build/versions/master/be/python.html

We would have stopped at that point, but it seemed helpful to file this issue because a) software other than Bazel may write under CDK’s `init-templates` directory, b) the errors that occur in this case are fairly obscure, and c) it should be fairly cheap to safely handle this case in CDK.
